metaflow/plugins/argo/argo_live_run.py

```diff
@@ -194,36 +200,37 @@ def _find_metaflow_run(self) -> None:
 
         namespace(None)
         metaflow_run_location = self._flow_name + "/" + self._metaflow_run_id
-        seconds_to_find_metaflow_run = 5  # gives 5 seconds to find MF run
+        seconds_to_find_metaflow_run = 5  # gives time to find MF run
         start_time = time.time()
-        print("Trying to find metaflow run")
+        print("Trying to find Metaflow run")
         print(f"metaflow_run_location: {metaflow_run_location}")
         while seconds_to_find_metaflow_run > time.time() - start_time:
-            print("Trying to find Metaflow Run")
             try:
                 metaflow_run = Run(metaflow_run_location)
                 self._metaflow_run = metaflow_run
                 print("Found it!")
                 break
-            except:
-                print("could not find metaflow Run")
+            except MetaflowNotFound:
                 time.sleep(1)
         else:
-            # TODO: add specificity to exceptions (see AIP-6470).
-            # MetaflowException (or a child exception in file)
-            # or TimeoutError would be more specific.
-            raise Exception("Unable to find Metaflow Run")
+            print("Failed to find Metaflow run")
+            if time.time() - self._time_triggered < 60:
+                # do not raise exception if still within 60s of triggering
+                print("Not raising exception because within 60 seconds of triggering")
+                return
+
+            raise TimeoutError(f"""
+    Unable to find Metaflow Run within {seconds_to_find_metaflow_run} seconds
+    Attempted to find at location: {metaflow_run_location}
+            """)
```

Comment:
Sorry for not catching this earlier, but all metaflow operations will happen within argo's knowledge. Instead of waiting for 5 seconds or relying on time triggered, you can simply check `if self._error`. `self.exception` shouldn't be called when the run has not finished.

Reply:
Solved in https://github.com/zillow/metaflow/pull/189

Reply:
This is resolved in next pr